04_basic_struct: Basic structure homework

Rampopula opened the pull request:
GL 04_basic_struct kernel loadable module task and solution

ekovalyov committed:
homework: basic data structures

Rampopula committed:
04_basic_struct: Add setup environment script

Added setup.sh script which setup enivornment variables to build kernel module under BeagleBone board

Signed-off-by: Eduard Chaika <rampopula@gmail.com>

alexposukhov commented on 04_basic_struct/hello_list.c:
```diff
@@ -0,0 +1,137 @@
+}
+
+static ssize_t hello_show(struct kobject *kobj, struct kobj_attribute *attr,
+char *buf)
```
Please add spaces to provide some indentation

Rampopula committed:
04_basic_struct: Basic structure homework

Kernel loadable module which creates a kobject with representation in the sysfs virtual filesystem
Kobject contains sysfs attribute with name 'list' which implements linked-list with read/write interface

Signed-off-by: Eduard Chaika <rampopula@gmail.com>
